### PR TITLE
added connection, chat management and contact with the server

### PR DESCRIPTION
**Pull request by Shenrak**
Features : 
* Added login page
* Added token management
* Added identification via token for the chat

Core : 
* Added utils to handle request responses
* Added reducers to handle informations about characters

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
     return (
-      <MuiThemeProvider>
+      <div>
+        <Overlay />
+        {!this.props.connected ? <Login /> : ""}
         <h1>GOR Project</h1>
         <div className="game">
-          <Chatbox />
-          {/* <Character /> */}
+          {this.props.connected ? <Chatbox /> : ""}
```
Comment (ViGrad) [inline src/App.jsx:20]: I think it could be better with a router or something like this

File: src/App.jsx [hunk 2]
```diff
@@ -1,24 +1,37 @@
...
         </div>
-      </MuiThemeProvider>
+      </div>
     )
   }
 }
 
-export default App
+const mapStateToProps = state => ({
+  connected: state.context.connected,
```
Comment (ViGrad) [inline src/App.jsx:30]: I advise you to use getters to get property from state, thereby components won't be impacted if you modify the state's shape

File: src/actions/index.js [hunk 3]
```diff
@@ -1,7 +1,39 @@
...
 export const addMessage = (text, owner) => ({
-  type: 'ADD_MESSAGE',
+  type: "ADD_MESSAGE",
+  id: nextMessageId++,
+  text,
+  owner
+})
+
+export const login = form => dispatch => {
+  dispatch(startLoading())
```
Comment (ViGrad) [inline src/actions/index.js]: With a control before if the app is already loading something, you can prevent your app to send multiple requests if some user press a lot of time on enter key

File: src/actions/loading.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import * as actionTypes from "./types"
+
+export const startLoading = (message = "") => dispatch => {
+  dispatch({
```
Comment (ViGrad) [inline src/actions/loading.js]: You don't have to use dispatch if your action only dispatch something

Comment (Shenrak) [reply src/actions/loading.js]: Nice

**Pull request merged**